storage_controller/src/service.rs

```diff
@@ -5055,6 +5055,69 @@ impl Service {
         Ok(TenantShardMigrateResponse {})
     }
 
+    pub(crate) async fn tenant_shard_migrate_secondary(
+        &self,
+        tenant_shard_id: TenantShardId,
+        migrate_req: TenantShardMigrateRequest,
+    ) -> Result<TenantShardMigrateResponse, ApiError> {
+        let waiter = {
```

Comment:
nit: is the block only to drop the lock? Maybe consider using `drop()` instead to avoid the indentation. 🤷

Reply:
I kind of like having blocks that show which part of a function is touching the synchronous-locked state, although in the case of these migrate functions it is a bit unwieldy.  This pattern exists in quite a few functions here, so I'm going to leave this as it is for consistency, although I'm totally open to discussing a broader refactor

Reply:
Yeah, totally fine, doesn't matter.